### PR TITLE
feat: grafana支持 all 选项 --story=115886431

### PR DESCRIPTION
**Pull request by TangMoonX**
# Reviewed, transaction id: 2114

### CODE REVIEW

File: bkmonitor/packages/monitor_web/grafana/resources/unify_query.py [hunk 1]
```diff
@@ -718,6 +718,8 @@ def perform_request(self, params):
 
             if query_config["interval"] == "auto":
                 query_config["interval"] = get_auto_interval(60, params["start_time"], params["end_time"])
+            # 删除全选条件
+            query_config["where"] = remove_all_conditions(query_config["where"])
```
Comment (TangMoonX) [inline bkmonitor/packages/monitor_web/grafana/resources/unify_query.py:718]: 自测截图：
1、返回所有数据
image

image

image

image

2、删除__ALL__条件，返回其他条件查询的结果
image

image

File: bkmonitor/packages/monitor_web/grafana/resources/unify_query.py [hunk 2]
```diff
@@ -1107,6 +1109,9 @@ def perform_request(self, params):
         else:
             start_time = time_interval_align(params["start_time"], interval)
             end_time = time_interval_align(params["end_time"], interval)
+            # 删除全选条件
+            params["promql"] = re.sub(r"[a-zA-Z_][a-zA-Z0-9_]*=['\"]__ALL__['\"]\s*,?", '', params["promql"])
+            params["promql"] = re.sub(r',\s*}', r'}', params["promql"])
```
Comment (TangMoonX) [inline bkmonitor/packages/monitor_web/grafana/resources/unify_query.py]: 自测截图：
1、返回所有数据
image

2、删除__ALL__条件，返回其他条件查询的结果
image

image

image

Comment (chenguo367) [reply bkmonitor/packages/monitor_web/grafana/resources/unify_query.py]: 仪表盘， promql模式下， 支持变量的选择，条件是会被设置成 =~ 或者 =

File: bkmonitor/packages/monitor_web/grafana/resources/unify_query.py [hunk 3]
```diff
@@ -1107,6 +1109,9 @@ def perform_request(self, params):
         else:
             start_time = time_interval_align(params["start_time"], interval)
             end_time = time_interval_align(params["end_time"], interval)
+            # 删除全选条件
+            params["promql"] = re.sub(r"[a-zA-Z_][a-zA-Z0-9_]*=['\"]__ALL__['\"]\s*,?", '', params["promql"])
```
Comment (chenguo367) [inline bkmonitor/packages/monitor_web/grafana/resources/unify_query.py]: 测试用例:
```
avg by (bk_target_ip) (bkmonitor:system:disk:in_use{bk_target_ip="__ALL__"})
avg by (bk_target_ip) (bkmonitor:system:disk:in_use{bk_target_ip='__ALL__'})
avg by (bk_target_ip) (bkmonitor:system:disk:in_use{bk_target_ip =  "__ALL__"  })
avg by (bk_target_ip) (bkmonitor:system:disk:in_use{bk_target_ip =  '__ALL__'  })
avg by (bk_target_ip) (bkmonitor:system:disk:in_use{bk_target_ip=~"__ALL__"})
avg by (bk_target_ip) (bkmonitor:system:disk:in_use{bk_target_ip =~ "__ALL__" })
avg by (bk_target_ip) (bkmonitor:system:disk:in_use{bk_target_ip=~"__ALL__" , bk_target_cloude='__ALL__' })
```

**Review by chenguo367 — CHANGES_REQUESTED** [review]

File: bkmonitor/packages/monitor_web/grafana/utils.py [hunk 4]
```diff
@@ -157,3 +157,35 @@ def get_cookies_filter() -> Optional[Dict]:
...
+    remove_all = is_remove_all(where_list, select_all_tag)
+    if remove_all:
+        return []
+    # 删除全选条件
+    final_where_list = [condition for condition in where_list if select_all_tag not in condition["value"]]
+
+    if final_where_list and "condition" in final_where_list[0]:
+        del final_where_list[0]["condition"]
+
+    return final_where_list
```
Comment (chenguo367) [inline bkmonitor/packages/monitor_web/grafana/utils.py]: 思考：
```
1. 多个条件的and 和 or连接
ip="__ALL__ " and ip="10.0.0.1"
ip="__ALL__ " or ip="10.0.0.1"
2. 单个条件的肯定与否定逻辑计算
ip="__ALL__ "
ip!="__ALL__ "  # 等于查不出数据（这种情况可以忽略，因为正常不太会这么勇）
# 重点关注第一种情况
```

**Review by chenguo367 — CHANGES_REQUESTED** [review]

File: bkmonitor/packages/monitor_web/grafana/resources/unify_query.py [hunk 5]
```diff
@@ -1107,6 +1109,9 @@ def perform_request(self, params):
         else:
             start_time = time_interval_align(params["start_time"], interval)
             end_time = time_interval_align(params["end_time"], interval)
+            # 删除全选条件
+            params["promql"] = re.sub(r"[a-zA-Z_][a-zA-Z0-9_]*\s*(=|=~)\s*['\"]__ALL__['\"]\s*,?", "", params["promql"])
+            params["promql"] = re.sub(r",\s*}", r"}", params["promql"])
```
Comment (tang202388) [inline bkmonitor/packages/monitor_web/grafana/resources/unify_query.py:1114]: 自测截图：
1、avg by (bk_target_ip) (bkmonitor:system:disk:in_use{bk_target_ip="__ALL__"})    查询返回所有数据
image
2、avg by (bk_target_ip) (bkmonitor:system:disk:in_use{bk_target_ip='__ALL__'})   查询返回所有数据
image
3、avg by (bk_target_ip) (bkmonitor:system:disk:in_use{bk_target_ip =  "__ALL__"  })  查询返回所有数据
image
4、avg by (bk_target_ip) (bkmonitor:system:disk:in_use{bk_target_ip =  '__ALL__'  })  查询返回所有数据
image
5、avg by (bk_target_ip) (bkmonitor:system:disk:in_use{bk_target_ip=~"__ALL__"})  查询返回所有
image
6、avg by (bk_target_ip) (bkmonitor:system:disk:in_use{bk_target_ip =~ "__ALL__" })   查询返回所有
image
7、avg by (bk_target_ip) (bkmonitor:system:disk:in_use{bk_target_ip=~"__ALL__" , bk_target_cloude='__ALL__' })   查询返回所
image
8、avg by (bk_target_ip) (avg_over_time(bkmonitor:system:load:load1{bk_target_ip="*.*.*.*",bk_target_ip="__ALL__"}[1m]))  查询返回 bk_target_ip="*.*.*.*"的值
image

Comment (TangMoonX) [inline bkmonitor/packages/monitor_web/grafana/resources/unify_query.py]: 1、avg by (bk_target_ip) (bkmonitor:system:disk:in_use{bk_target_ip="\_\_ALL\_\_"})  查询返回所有
image
2、avg by (bk_target_ip) (bkmonitor:system:disk:in_use{bk_target_ip='\_\_ALL\_\_'})  查询返回所有
image
3、avg by (bk_target_ip) (bkmonitor:system:disk:in_use{bk_target_ip =  "\_\_ALL\_\_"  })  查询返回所有
image
4、avg by (bk_target_ip) (bkmonitor:system:disk:in_use{bk_target_ip =  '\_\_ALL\_\_'  })   查询返回所有
image
5、avg by (bk_target_ip) (bkmonitor:system:disk:in_use{bk_target_ip=~"\_\_ALL\_\_"})   查询返回所有
image
6、avg by (bk_target_ip) (bkmonitor:system:disk:in_use{bk_target_ip =~ "\_\_ALL\_\_" })  查询返回所有
image
7、avg by (bk_target_ip) (bkmonitor:system:disk:in_use{bk_target_ip=~"\_\_ALL\_\_" , bk_target_cloude='\_\_ALL\_\_' })  查询返回所有
image
8、avg by (bk_target_ip) (avg_over_time(bkmonitor:system:load:load1{bk_target_ip="*.*.*.*",bk_target_ip="\_\_ALL\_\_"}[1m]))  查询返回 bk_target_ip="*.*.*.*"的值
image

**Comment by unique0lai** [comment]
考虑到使用场景的特殊性，只有在特定条件下触发该机制
1. promql的情况下，当method为~=或=，value为__ALL__时，剔除该条件
2. 结构化参数查询的情况下，当method为肯定条件(eq/include/regex)时，剔除该条件，如果剔除该条件导致逻辑恒等，则消除其他条件。

**Review by unique0lai — DISMISSED** [review]

**Review by chenguo367 — APPROVED** [review]

**Pull request merged**